ts/features/zendesk/screens/__tests__/ZendeskChooseCategory.test.tsx

```diff
@@ -0,0 +1,178 @@
+import { NavigationParams } from "react-navigation";
+import { createStore, Store } from "redux";
+import { fireEvent, RenderAPI } from "@testing-library/react-native";
+import { appReducer } from "../../../../store/reducers";
+import { applicationChangeState } from "../../../../store/actions/application";
+import { GlobalState } from "../../../../store/reducers/types";
+import { renderScreenFakeNavRedux } from "../../../../utils/testWrapper";
+import ROUTES from "../../../../navigation/routes";
+import * as zendeskAction from "../../store/actions";
+import { getZendeskConfig } from "../../store/actions";
+import { Zendesk } from "../../../../../definitions/content/Zendesk";
+import ZendeskChooseCategory from "../ZendeskChooseCategory";
+import { ZendeskCategories } from "../../../../../definitions/content/ZendeskCategories";
+import { ZendeskSubCategories } from "../../../../../definitions/content/ZendeskSubCategories";
+import { ZendeskCategory } from "../../../../../definitions/content/ZendeskCategory";
+import MockZendesk from "../../../../__mocks__/io-react-native-zendesk";
+import * as navigationAction from "../../store/actions/navigation";
+import * as mixpanel from "../../../../mixpanel";
+
+jest.useFakeTimers();
+
+const mockedZendeskConfig: Zendesk = {
+  panicMode: false,
+  zendeskCategories: {
+    id: "12345",
+    categories: [
+      {
+        value: "mockedValue",
+        description: {
+          "it-IT": "mockedDescription",
+          "en-EN": "mockedDescription"
+        }
+      }
+    ]
+  }
+};
+
+const mockedZendeskSubcategories: ZendeskSubCategories = {
+  id: "67890",
+  subCategories: [
+    {
+      value: "mockedSubcategoryValue",
+      description: {
+        "it-IT": "mockedSubcategorydescription",
+        "en-EN": "mockedSubcategorydescription"
+      }
+    }
+  ]
+};
+
+describe("the ZendeskChooseCategory screen", () => {
+  const zendeskSupportFailureSpy = jest.spyOn(
+    zendeskAction,
+    "zendeskSupportFailure"
+  );
+  const globalState = appReducer(undefined, applicationChangeState("active"));
+
+  describe("should call the zendeskSupportFailure", () => {
```

Comment:
I'm unsure what to think about this methodology: it surely is a smart way of optimizing the use of `beforeEach`, `afterEach`, and `it`.
On the other hand, it only works because [the tests run sequentially](https://jestjs.io/docs/setup-teardown#order-of-execution-of-describe-and-test-blocks):

> Once the describe blocks are complete, by default Jest runs all the tests serially in the order they were encountered in the collection phase, waiting for each to finish and be tidied up before moving on.

I see the following issues:

1. it will force us to perform a refactoring if and when parallelization will be a thing
2. if the test at line [69](https://github.com/pagopa/io-app/pull/3674/files#diff-4d96a34d14e47210e9c9ead4d2a2d88a471791d50ac3c67aed97728b19b88bf0R69) fails, one has to jump to line [86](https://github.com/pagopa/io-app/pull/3674/files#diff-4d96a34d14e47210e9c9ead4d2a2d88a471791d50ac3c67aed97728b19b88bf0R86) and  [62](https://github.com/pagopa/io-app/pull/3674/files#diff-4d96a34d14e47210e9c9ead4d2a2d88a471791d50ac3c67aed97728b19b88bf0R62) just to understand what is the test about.
3. any update to a single test (or adding a new one) will require the developer to understand how come they share variables but are not interdependent - it took me some 10 minutes 😓  

When it comes to testing, I'd rather be WET than DRY 😄 to maximize isolation over code reuse.

Reply:
You convinced me!
It seemed interesting to me as an approach but already now that I looked at it during the correction it seemed confusing to me.
Refactored in [817ba12](https://github.com/pagopa/io-app/pull/3674/commits/817ba1277e2156c3578002fc13b31a4f038c49f2)